src/app/components/layout/app-layout/app-layout.component.ts

```diff
@@ -137,18 +137,9 @@ export class AppLayoutComponent implements OnInit, OnDestroy {
         this.updateState();
       });
 
-    this.router.events
-      .pipe(
-        filter(event => {
-          return (
-            event instanceof NavigationStart &&
-            // search employs reuse route strategy
-            !event.url.startsWith('/search;')
-          );
-        }),
-        takeUntil(this.onDestroy$)
-      )
-      .subscribe(() => this.store.dispatch(new SetSelectedNodesAction([])));
+    this.router.events.pipe(takeUntil(this.onDestroy$)).subscribe(() => {
+      this.store.dispatch(new ResetSelectionAction());
```

Comment:
Do we need to reset selection on each of the router events? We should filter the router events like it was before